src/main/java/ui/UI.java

```diff
@@ -239,6 +242,30 @@ private void initUI(Stage stage) {
         loadFonts();
         String css = initCSS();
         applyCSS(css, scene);
+
+        setApplicationIcon(stage);
+    }
+
+    public void setApplicationIcon(Stage stage) {
+        stage.getIcons().add(new Image(UI.class.getResourceAsStream("logo.png")));
```

Comment:
Is this bit needed for OS X as well?

Also maybe pull out the image file name into a constant.